### DISCUSSION TITLE
How to format only files inside current workspace?

### DISCUSSION THREAD

**Discussion by t1m0thyj** [prettier/prettier-vscode#3303] [Q&A]
For any files inside my VS Code project folder, I want Prettier to auto-format them on save. So I have enabled the setting `editor.formatOnSave` at the workspace level (in ".vscode/settings.json" inside my project).

When I open a file from outside my workspace in the same VS Code window as my project, then it gets formatted on save too which I don't want. How can I prevent this?

**Comment by Labeeb2339** [comment]
Add this to `.vscode/settings.json`:

{
  "editor.formatOnSave": true,
  "prettier.requireConfig": true
}

`prettier.requireConfig` makes the extension format only files with a Prettier configuration, so files outside the project without one won’t be formatted. See the [official setting documentation](https://github.com/prettier/prettier-vscode#prettierrequireconfig-default-false).
